internal/olm/client/status.go

```diff
@@ -93,9 +93,11 @@ func (s Status) HasInstalledResources() (bool, error) {
 	sort.Slice(s.Resources, func(i int, j int) bool {
 		return s.Resources[i].Resource != nil
 	})
+	existOLM := false
 	for _, r := range s.Resources {
 		if r.Resource != nil {
-			return true, nil
+			log.Infof("	Should remove exist OLM resource： Kind:%s, Name:%s", r.GVK.Kind, r.NamespacedName.Name)
+			existOLM = true
 		} else if r.Error != nil && !apierrors.IsNotFound(r.Error) {
```

Comment:
it shows that will not work as expected. note that the func will never more be going inside here. 
Also, I think we have 2 needs/2 prs or 2 fragments in this PR. One to increase the timeout other to change the bahaviour of the status command.

Reply:
> it shows that will not work as expected. note that the func will never more be going inside here.

Yes, I think so. Remove it?

> Also, I think we have 2 needs/2 prs or 2 fragments in this PR. One to increase the timeout other to change the bahaviour of the status command.

Yes, do I need to split them?

Reply:
I think if you do a PR for the timeout and another to change the status we could merge the timeout and then working in the second one. Regards the implementation here it shows not accurate. If we installed the OLM with the operator-sdk will not `if r.Resource != nil` always true? did you test that?